doc/tutorial/read_mapper/index.md

```diff
@@ -0,0 +1,236 @@
+# Implementing your own read mapper with SeqAn3 {#tutorial_read_mapper}
+
+<b>Learning Objective:</b><br>
+In this tutorial you will learn how to combine the components of previous tutorials to create your very first
+SeqAn3 application: a read mapper!
+
+\tutorial_head{High, 90 Minutes, All,}
+
+[TOC]
+
+<br><br>
+
+# Introduction
+
+Read mapping is a common task in bioinformatics and is often the first step of an in-depth analysis of
+[Next Generation Sequencing](https://en.wikipedia.org/wiki/DNA_sequencing#High-throughput_methods) data. Its aim
+is to identify positions where a query sequence (read) matches with up to `e` errors to a reference sequence.
+
+In this example we will implement a read mapper step by step and make use of what we have learned in
+the previous tutorials.
+As it is common practice with read mappers, we will first create a indexer that creates an index from the reference
+and stores it to disk. After this, we will implement the actual read mapper that will use the stored index and
+map the reads.
+
+# Agenda
+
+* Implementing an indexer
+    * Parse arguments
+    * Read input files
+    * Create and store index
+* Implementing a read mapper
+    * Parse arguments
+    * Read and load input, search for approximate matches
+    * Align the search results
+    * Write final results into a SAM file
+
+# The indexer
+
+## Step 1 - Parsing arguments
+As a first step, we want to parse command line arguments for our indexer. If you get into trouble,
+you can take a peek at the \ref tutorial_argument_parser "Argument Parser Tutorial" or the API documentation of
+the seqan3::argument_parser for help.
+
+\assignment{Parsing arguments}
+Let's start our application by setting up the argument parser with the following options:
+* The path to the reference file
+* An output path for the index
+
+Follow the best practice and create:
+* A function `run_program` that prints the parsed arguments
+* A struct `cmd_arguments` that stores the arguments
+* A function `initialise_argument_parser` to add meta data and options to the parser
+* A `main` function that parses the arguments and calls `run_program`
+
+Use validators where applicable!
+
+Your `main` may look like this:
+\hint
+\snippet doc/tutorial/read_mapper/read_mapper_indexer_step1.cpp main
+\endhint
+\endassignment
+\solution
+\include doc/tutorial/read_mapper/read_mapper_indexer_step1.cpp
+\endsolution
+
+## Step 2 - Reading the input
+As a next step, we want to use the parsed file names to read in our data. This will be done for the reference
```

Comment:
```suggestion
As a next step, we want to use the parsed file name to read in our reference data. This will be done 
```
sorry :D since it is only one now I would change it back